resources/metric-ui.mjs

```diff
@@ -0,0 +1,292 @@
+export const COLORS = Object.freeze(["blue", "blue-light", "green-light", "green", "yellow", "orange", "red", "magenta", "violet", "purple", "blue-dark", "green-dark", "ochre", "rust"]);
+
+export function renderMetricView(params) {
+    let { metrics, width = 500, trackHeight = 20, subMetricMargin = 35, title = "", colors = COLORS} = params;
+    // Make sure subMetricMargin is set for use in renderSubMetrics.
+    params.subMetricMargin = subMetricMargin;
+    const scatterPlotParams = { width, trackHeight, colors };
+    scatterPlotParams.values = prepareScatterPlotValues(metrics, true);
+    scatterPlotParams.unit = "%";
+    scatterPlotParams.xAxisLabel = "Spread Normalized";
+    const normalizedScatterPlot = renderScatterPlot(scatterPlotParams);
+    scatterPlotParams.values = prepareScatterPlotValues(metrics, false);
+    scatterPlotParams.unit = metrics[0].unit;
+    scatterPlotParams.xAxisLabel = metrics[0].unit;
+    const absoluteScatterPlot = renderScatterPlot(scatterPlotParams);
+    const legend = metrics
+        .map(
+            (metric, i) => `
+                <tr >
+                    <td class=${colors[i % colors.length]} >●</td>
```

Comment:
Are we sure all these values come from trusted sources and not via query parameters, etc...?
I don't want us to have XSS vulnerabilities.

Reply:
- I've  now renamed `params` to `viewParams` to make a bit more clear that they are not related.
- params.mjs parses all input data (except for `params.suites` ) to int and bool
- params.suites is checked against a known list of Suites.keys() in runBenchmark

The input vaues to renderMetricView are only extracted from the metrics themselves which have values coming only from the runner.